engine/interpreter/src/main/scala/pl/touk/nussknacker/engine/definition/ServiceInvoker.scala

```diff
@@ -66,10 +66,8 @@ object ServiceInvoker {
 
   def apply(objectWithMethodDef: ObjectWithMethodDef, collector: Option[ServiceInvocationCollector] = None): ServiceInvoker = {
     val detectedReturnType = (objectWithMethodDef.methodDef.realReturnType match {
-      case Typed(set) =>
-        //FIXME: this headOption assumes that there will be only one class here
-        //in the future we should get rid of Set in Typed...
-        set.headOption.map(_.klass)
+      case tc: ScalarTypingResult =>
```

Comment:
I wonder if we can make methodDef.realReturnType just be TypedClass... For now let's leave it like this